### PR TITLE
Adding support for scoreboard channel specific timeouts

### PR DESCRIPTION
**Pull request by Intuity**

### CODE REVIEW

File: docs/components.md [hunk 1]
```diff
@@ -217,6 +217,90 @@ If, for any reason, you do not want a monitor to be attached to the scoreboard
 then you may provide the argument `scoreboard=False` to the `self.register(...)`
 call.
 
+### Scoreboard Channel Types
+
+When a monitor is attached onto the scoreboard (i.e. `self.register(...)` is
```
Comment (Alldred) [inline docs/components.md:222]: This is a confusing sentence to read

File: docs/components.md [hunk 2]
```diff
@@ -217,6 +217,90 @@ If, for any reason, you do not want a monitor to be attached to the scoreboard
...
+to scoreboard channels that specify how long it is acceptable for a transaction
+to sit at the front of the monitor's queue before the scoreboard matches it to
+a reference transaction. There are two key parameters to `self.register(...)`:
+
+ * `scoreboard_timeout_ns` - the maximum acceptable age a transaction may be at
+   the front of the monitor's channel. The age is determined by substracting the
+   transaction's `timestamp` field (a default property of `BaseTransaction`)
+   from the current simulation time.
+
+ * `scoreboard_polling_ns` - the frequency with which to check the front of the
```
Comment (Alldred) [inline docs/components.md:284]: Is it worth making this a percentage of the timeout? (ie large timeouts could cope with larger polling values).
With a minimum so it doesn't get silly for performance?

File: docs/components.md (hunk 1 above)
Comment (Intuity) [reply docs/components.md]: Any better now?

File: docs/components.md (hunk 2 above)
Comment (Intuity) [reply docs/components.md:285]: I think it's fine - quite possibly in future we'd want to do other things with the polling loop, so having it as a standalone value is probably useful.

**Review by Alldred — APPROVED** [review]

File: docs/components.md (hunk 1 above)
Comment (Alldred) [reply docs/components.md]: Much! :-)

**Pull request merged**